lib/new_relic/agent/configuration/manager.rb

```diff
@@ -16,6 +16,8 @@ module Agent
     module Configuration
       class Manager
 
+        CONFIG_WITH_HASH_VALUE = ['expected_messages', 'ignore_messages']
+
```

Comment:
I think this is a place where more detail could be helpful. What about a comment like this? 

```
        # These are configuration options that have a value of a Hash
        # This is used in YamlSource#dot_flattened prevent flattening these values
```